### PR TITLE
lz4: update to release version v1.10.0

### LINKED ISSUE #71970: LZ4 compression sample should explain how to install optional lz4 module
**Describe the bug**
Building lz4 sample seems broken after v3.4.0 regardless of platform
Build complains of missing lz4.h
```
/home/user/zephyrproject/zephyr/samples/compression/lz4/src/main.c:11:10: fatal error: lz4.h: No such file or directory
   11 | #include "lz4.h"
      |          ^~~~~~~
compilation terminated.
```
Git bisect point to the commit f30f08a3ce75ae7d7384a7608b7e3dcef804661a where submodules were made optional

**To Reproduce**
1. cd ~/zephyrproject/zephyr/samples/compression/lz4
2. west build -b qemu_x86 --pristine
3. See error

Additionally the cmake phase prints:
```
warning: LZ4 (defined at modules/lz4/Kconfig:7) was assigned the value 'y' but got the value 'n'.
Check these unsatisfied dependencies: 0 (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_LZ4 and/or look up LZ4 in the
```
This warning is pretty much useless. What is dependency 0? How to deal with it?
In menuconfig interface it is impossible to enable LZ4 option either

**Expected behavior**
If the optional module needs to be somehow enabled, a message should be displayed in kconfig phase (probably by config script)

**Impact**
So far I could not find a way to build this sample in v3.5.0+. Nothing that I could see in the documentation either.

**Logs and console output**
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.16.5.1
 - v3.5.0

### PR DESCRIPTION
**Pull request by parthitce**
LZ4 is a fast compression and only of it's kind available in Zephyr RTOS manifest. But it wasn't maintained to sync with mainline version. This commit together with [1] PR help updating to latest mainline release version of LZ4.

[1]: https://github.com/zephyrproject-rtos/lz4/pull/2

### CODE REVIEW

**Comment by parthitce** [comment]
> @parthitce [zephyrproject-rtos/lz4#2](https://github.com/zephyrproject-rtos/lz4/pull/2) is merged, please replace with SHA.

Thanks, done.

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -4798,9 +4798,10 @@ West:
 "West project: lz4":
   status: odd fixes
   collaborators:
-    - Navin-Sankar
+    - parthitce
```
Comment (MaureenHelm) [inline MAINTAINERS.yml:4801]: @Navin-Sankar please confirm

Comment (Navin-Sankar) [reply MAINTAINERS.yml:4801]: Yes, I confirm

**Review by Navin-Sankar — APPROVED** [review]
LGTM.

**Review by MaureenHelm — APPROVED** [review]

**Review by mniestroj — APPROVED** [review]

**Pull request merged**